src/imitation/algorithms/dagger.py

```diff
@@ -310,25 +358,39 @@ def _check_has_latest_demos(self):
                 f".get_trajectory_collector()"
             )
 
-    def _try_load_demos(self):
-        self._check_has_latest_demos()
         if self._last_loaded_round < self.round_num:
+            # TODO(shwang): Do we actually only want to load the most recent round's
+            #   data, or should we load all the data?
             transitions, num_demos = self._load_all_demos()
             logging.info(
                 f"Loaded {sum(num_demos)} new demos from {len(num_demos)} rounds"
             )
             data_loader = th_data.DataLoader(
                 transitions,
                 self.batch_size,
-                drop_last=True,
+                # If drop_last=True, then BC.train() fails on "no batches available"
```

Comment:
Yes, this is why I always set `drop_last=True`.

Reply:
It's annoying to fail due to not enough samples, because the error message you get is not at all obvious (something like "BC data loader yielded no batches unexpected"). Also, testing becomes a pain for both DAgger and the associated Sacred script because you have to set a high enough number of timesteps to ensure we don't run into this issue while still trying to shorten test duration.

I'm going to revert to `drop_last=True` as recommended, and then ensure that `train()` doesn't end a round without collecting `self.batch_size` samples, and also add a big `assert` or `RuntimeError` here if we still don't have enough samples for a single batch.